File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }
        StringBuffer sb = new StringBuffer();
        Stack<Integer> stack = new Stack<>();
        int num = 1;
        boolean print = false;

        for (int i = 0; i < N; i++) {
            if (A[i] >= num) {
                while (A[i] >= num) {
                    stack.push(num++);
                    sb.append("+\n");
                }
                stack.pop();
                sb.append("-\n");
            } else {
                int pop = stack.pop();
                if (pop > A[i]) {
                    System.out.println("NO");
                    print = true;
                    break;
                } else sb.append("-\n");
            }
        }
        if (!print) System.out.println(sb.toString());
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include <iostream>
#include <stack>
#include <vector>
#include <string>
using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	int N;
	cin >> N;

	stack<int> stack;
	int num = 1, n;
	string output;
	for (int i = 0; i < N; i++)
	{
		cin >> n;

		while (num <= n)
		{
			stack.push(num++);
			output += "+\n";
		}

		if (stack.top() == n)
		{
			stack.pop();
			output += "-\n";
		}
		else if (stack.top() > n)
		{
			cout << "NO";
			return 0;
		}
	}

	cout << output;

}